swayfmt/src/formatter/shape.rs

```diff
@@ -194,22 +214,9 @@ impl Shape {
     // |<---->|          indent
     //        |<--->|    width
     //
-    /// Create a copy of
-    pub(crate) fn from(shape: &Shape, width: Option<usize>, code_line: Option<CodeLine>) -> Self {
-        let mut new_shape = *shape;
-        if width.is_some() {
-            new_shape.update_width(width.unwrap_or_default());
-        }
-        if let Some(code_line) = code_line {
-            new_shape.update_line_settings(code_line)
-        }
-
-        new_shape
-    }
     /// A wrapper for `to_width_heuristics()` that also checks if the settings are default.
     pub(crate) fn apply_width_heuristics(&mut self, width_heuristics: WidthHeuristics) {
-        if width_heuristics == WidthHeuristics::default() {
-        } else {
+        if width_heuristics != WidthHeuristics::default() {
             self.width_heuristics = width_heuristics
```

Comment:
Just noticed this - are we sure we want to deny users of this method from resetting this value back to the default state? Perhaps we can just remove the condition altogether and let the function body be `self.width_heuristics`? Let me know if I'm missing the point or use-case here!

Also, the method mentions it's a wrapper for `to_width_heuristics()`, but it appears there's no call to `to_width_heuristics()`.

Reply:
So the heuristic settings are what we gather from the `swayfmt.toml`, I don't believe we would want this to change during formatting. This function just takes in the settings that are generated from `to_width_heuristics`. If I recall correctly, I attempted to make it a direct application but this is the conclusion I came to at the time.

Reply:
To be more clear, at the beginning of each `Formatter::format` call, we are reassigning the `heuristics` settings from a `config`. This is just the method that applies those settings to `Shape`.